tsl/src/chunk_api.c

```diff
@@ -1187,9 +1260,12 @@ fetch_remote_chunk_stats(Hypertable *ht, FunctionCallInfo fcinfo, bool col_stats
 						"that cannot accept type record")));
 
 	cmdres = ts_dist_cmd_invoke_func_call_on_all_data_nodes(fcinfo);
+
 	/* Expect TEXT response format since dist command API currently defaults
 	 * to requesting TEXT */
 	tf = tuplefactory_create_for_tupdesc(tupdesc, true);
+	num_rows = ts_dist_cmd_total_row_count(cmdres);
+	stats_process_context_init(&statsctx, num_rows / ht->fd.replication_factor);
```

Comment:
There is a potential rounding error here due to division. You should use ceil(num_row/ht->fd_replication_factor). Or optimize by computing ceil(m/n) as (m+n-1)/n
The other option is we know exactly how many rows are expected. For relstats: 1 for each chunk, so this is equal to num_chunks . For colstats, it is num_chunks * num_cols?Is there a reason why we are not initializing using this approach?

Reply:
Not sure rounding errors matters much and this is just to set the initial size of the hash table. It will automatically expand if necessary. If anything, I'd make the initial size a bit larger than necessary to avoid expanding the hash table.

I'll look into your second suggestion.

Reply:
Didn't know the table expanded if there are more elements. It is fine in that case.